integration/src/main/java/com/aws/greengrass/mqttbroker/ClientDeviceTrustManager.java

```diff
@@ -70,9 +70,19 @@ public String getSessionForCertificate(X509Certificate... x509Certificates) {
             return null;
         }
 
-        String sessionId = sessionMap.remove(certPem);
+        return getSessionForCertificate(certPem);
+    }
+
+    /**
+     * Returns a valid session ID for the given certificate chain, if it exists.
+     *
+     * @param certificatePem peer certificate PEM
+     * @return a session id
+     */
+    public String getSessionForCertificate(String certificatePem) {
+        String sessionId = sessionMap.remove(certificatePem);
```

Comment:
Why are you removing and not just getting?

Reply:
This is in the TrustManager, not the Authorizer. We're removing to pass ownership to the Authorizer, which will attach a Thing to the session. At that point, we'd want subsequent requests for the same certificate to get a different session. Consider the case where two things share the same cert.